packages/devextreme/js/ui/context_menu/ui.context_menu.js

```diff
@@ -603,25 +605,67 @@ class ContextMenu extends MenuBase {
         }
     }
 
+    _initScrollView($container, anchor, considerAnchorHeight = true) {
+        const containerHeight = getOuterHeight($container);
+        const maxHeight = this._getMaxHeight(containerHeight, anchor, considerAnchorHeight);
+        const menuHeight = Math.min(containerHeight, maxHeight);
+        const cssProps = {
+            height: menuHeight,
+            overflow: 'visible',
+        };
+        const containerOffset = $container.offset().top;
+
+        if(containerOffset < 0) {
+            cssProps.top = -containerOffset;
+        }
+        $container.css(cssProps);
+        this._createComponent($container, ScrollView, {});
+    }
+
+    _getMaxHeight(containerHeight, anchor, considerAnchorHeight) {
+        const windowHeight = getOuterHeight(window);
+        const isContextMenuRootEvent = anchor.type === DEFAULT_SHOW_EVENT || !anchor.offset;
+        const document = domAdapter.getDocument();
+        const isAnchorDocument = anchor.length && anchor[0] === document;
+
+        if(isContextMenuRootEvent || isAnchorDocument) {
+            return windowHeight;
+        }
+
+        const offsetTop = anchor.offset().top;
+        let maxHeight = Math.max(offsetTop, windowHeight - offsetTop);
```

Comment:
[mistake]
it's not correct to make `max` here and plus/minus anchor height later because this `+/-` can affect the result of this `max`. 
For example, now Menu's flipped 1st-level submenu height is incorrect